ImmuDB4Net/ImmuClient.cs

```diff
@@ -94,37 +90,99 @@ internal IConnection Connection
+
+    /// <summary>
+    /// Gets or sets the length of time interval between periodic checks of idle connections to be closed. 
+    /// </summary>
+    /// <value></value>
+    public TimeSpan IdleConnectionCheckInterval { get; internal set; }
 
-    public class LibraryWideSettings
+    /// <summary>
+    /// Gets the grpc address of the ImmuDB server
+    /// </summary>
+    /// <value></value>
+    /// <remarks>
+    /// This value is computed from the server url and server port arguments that come either from 
+    /// <see cref="ImmuClientBuilder.Build()" /> or <see cref="ImmuClient(string, int)" /> constructor
+    /// </remarks>
+    public string GrpcAddress { get; }
+
+    /// <summary>
+    /// Gets the <see cref="LibraryWideSettings" /> object with the process wide settings
+    /// </summary>
+    /// <value></value>
+    public static LibraryWideSettings GlobalSettings
     {
-        public int MaxConnectionsPerServer { get; set; }
-        public TimeSpan TerminateIdleConnectionTimeout { get; set; }
-        public TimeSpan IdleConnectionCheckInterval { get; set; }
-        internal LibraryWideSettings()
+        get
         {
-            //these are the default values
-            MaxConnectionsPerServer = 2;
-            TerminateIdleConnectionTimeout = TimeSpan.FromSeconds(60);
-            IdleConnectionCheckInterval = TimeSpan.FromSeconds(6);
+            return globalSettings;
         }
     }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="ImmuClientBuilder" /> factory object for <see cref="ImmuClient" /> instances
+    /// </summary>
+    /// <returns>A new instance of <see cref="ImmuClient" /></returns>
+    public static ImmuClientBuilder NewBuilder()
+    {
+        return new ImmuClientBuilder();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ImmuClient" />. It uses the default value of 'localhost:3322' as ImmuDB server address and port and 'defaultdb' as database name.
+    /// </summary>
+    /// <returns></returns>
     public ImmuClient() : this(NewBuilder())
     {
 
     }
 
+    /// <summary>
+    ///  Initializes a new instance of <see cref="ImmuClient" />. It uses the implicit 'defaultdb' value for the database name.
+    /// </summary>
+    /// <param name="serverUrl">The ImmuDB server address, ex: localhost or http://localhost </param>
+    /// <param name="serverPort">The port where the ImmuDB server listens</param>
     public ImmuClient(string serverUrl, int serverPort)
         : this(NewBuilder().WithServerUrl(serverUrl).WithServerPort(serverPort))
     {
     }
 
+    /// <summary>
+    /// Initializes a new instance of <see cref="ImmuClient" />
+    /// </summary>
+    /// <param name="serverUrl">The ImmuDB server address, ex: localhost or http://localhost </param>
+    /// <param name="serverPort">The port where the ImmuDB server listens</param>
+    /// <param name="database">The name of the ImmuDB server's database to use</param>
```

Comment:
Just noticed that - what's the use case for this constructor? We will have to specify the database name in `Open` anyway?

Reply:
This can use the default name 'defaultdb'. The aim was to make the use of this SDK as simple as possible, this was stated by our leadership.

Reply:
I can see that this could be a shorthand version, but I still can't find the use-case for it.

When opening a connection first the constructor is callde - so here we would provide the db name.
But then we have to call Open() again which requires db name. So why we ever need to specify the db name in constructor?

It would make sense to me if there was some alternative `Open` method that does not take the db name parameter and instead takes the one set in constructor but I didn't see one.

Reply:
Removed this overload.